history/src/modules/wifi/wifi_atks_20240802063120.cpp

```diff
@@ -0,0 +1,572 @@
+// Borrowed from https://github.com/justcallmekoko/ESP32Marauder/
+// Learned from https://github.com/risinek/esp32-wifi-penetration-tool/
+// Arduino IDE needs to be tweeked to work, follow the instructions: https://github.com/justcallmekoko/ESP32Marauder/wiki/arduino-ide-setup
+// But change the file in: C:\Users\<YOur User>\AppData\Local\Arduino15\packages\m5stack\hardware\esp32\2.0.9
+#include <Arduino.h>
+#include "core/globals.h"
+#include "core/display.h"
+#include "core/main_menu.h"
+#include "core/mykeyboard.h"
+#include "core/wifi_common.h"
+#include "wifi_atks.h"
+#include "esp_wifi.h"
+#include "esp_system.h"
+#include "evil_portal.h"
+#include "vector"
+
+std::vector<wifi_ap_record_t> ap_records;
+/**
+ * @brief Decomplied function that overrides original one at compilation time.
+ *
+ * @attention This function is not meant to be called!
+ * @see Project with original idea/implementation https://github.com/GANESH-ICMC/esp32-deauther
+ */
+extern "C" int ieee80211_raw_frame_sanity_check(int32_t arg, int32_t arg2, int32_t arg3)
+{
+  if (arg == 31337)
+    return 1;
+  else
+    return 0;
+}
+
+uint8_t deauth_frame[sizeof(deauth_frame_default)]; // 26 = [sizeof(deauth_frame_default[])]
+
+wifi_ap_record_t ap_record;
+
+/***************************************************************************************
+** Function: send_raw_frame
+** @brief: Broadcasts deauth frames
+***************************************************************************************/
+void wsl_bypasser_send_raw_frame(const uint8_t *frame_buffer, int size)
```

Comment:
change this function name to
`void send_raw_frame(const uint8_t *frame_buffer, int size)`

this will need to be changed in the evil_portal.cpp in [this line](https://github.com/pr3y/Bruce/blob/bb4b0f37cfcf34aec9becbd05ce04a0d0f4a9ae7/src/modules/wifi/evil_portal.cpp#L175)

and chenge the function name on the wifi_atks.h, adding the `void deauthFloodAttack();` function to it